File: code/server/jvm/im-http-api/src/main/kotlin/im/api/model/output/invitations/ChannelInvitationOutputModel.kt
package im.api.model.output.invitations

import im.api.model.output.channel.ChannelOutputModel
import im.api.model.output.users.UserOutputModel
import java.time.LocalDateTime

data class ChannelInvitationOutputModel(
    val id: Long,
    val channel: ChannelOutputModel,
    val inviter: UserOutputModel,
    val invitee: UserOutputModel,
    val status: String,
    val role: String,
    val expiresAt: LocalDateTime,
) {
    companion object {
        fun fromDomain(invitation: im.domain.invitations.ChannelInvitation): ChannelInvitationOutputModel =
            ChannelInvitationOutputModel(
                id = invitation.id.value,
                channel = ChannelOutputModel.fromDomain(invitation.channel),
                inviter = UserOutputModel.fromDomain(invitation.inviter),
                invitee = UserOutputModel.fromDomain(invitation.invitee),
                status = invitation.status.toString(),
                role = invitation.role.toString(),
                invitation.expiresAt,
            )
    }
}

File: code/client/web/src/Dto/output/invitations/ChannelInvitationOutputModel.ts
import { UserOutputModel } from '../users/UserOutputModel';
import { ChannelOutputModel } from '../channels/ChannelOutputModel';
import { ChannelInvitationStatus } from '../../../Domain/invitations/ChannelInvitationStatus';
import { ChannelRole } from '../../../Domain/channel/ChannelRole';

/**
 * Channel invitation output model
 *
 * @param id The unique identifier of the channel invitation.
 * @param channel The channel to which the user is invited.
 * @param invitee The user who is invited to the channel.
 * @param inviter The user who invited the invitee to the channel.
 * @param status The status of the invitation.
 * @param role The role of the user in the channel.
 * @param expiresAt The date and time when the invitation expires.
 */
export interface ChannelInvitationOutputModel {
    id: number;
    channel: ChannelOutputModel;
    invitee: UserOutputModel;
    inviter: UserOutputModel;
    status: ChannelInvitationStatus;
    role: ChannelRole;
    expiresAt: string;
}
